lib/pages/ui/nps_rating_bar.dart

```diff
@@ -0,0 +1,121 @@
+    return Center(
+      child: Column(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        mainAxisAlignment: MainAxisAlignment.center,
+        children: [
+          Container(
+            alignment: Alignment.center,
+            height: 56,
+            width: double.infinity,
+            child: ListView.builder(
+              shrinkWrap: true,
+              scrollDirection: Axis.horizontal,
+              itemCount: counter,
+              itemBuilder: (context, index) {
+                return GestureDetector(
+                  onTap: () {
+                    _onRateSelected(index);
+                  },
+                  child: Container(
+                    alignment: Alignment.center,
+                    width: 34,
+                    decoration: BoxDecoration(
+                      border: Border.all(
+                        color: (_selectedRate != null && _selectedRate >= index)
+                            ? Colors.white
+                            : Colors.white30,
+                        width: 1,
+                      ),
+                      borderRadius: _roundedBorder(index, counter),
+                    ),
+                    child: Text(
+                      (index + 1).toString(),
+                      style: style.copyWith(
+                        color: (_selectedRate != null && _selectedRate >= index)
+                            ? Colors.white
+                            : Colors.white30,
+                      ),
+                    ),
+                  ),
+                );
+              },
+            ),
+          ),
+          const SizedBox(height: 15),
+          Padding(
+            padding: EdgeInsets.symmetric(horizontal: 30),
+            child: Row(
+              mainAxisAlignment: MainAxisAlignment.spaceBetween,
+              children: [
+                Text(
+                  minTitle,
+                  style: style.copyWith(
+                    color:
+                        (_selectedRate != null && _selectedRate >= counter / 2)
+                            ? Colors.white30
+                            : Colors.white,
+                  ),
+                ),
+                Text(maxTitle,
+                    style: style.copyWith(
+                      color: (_selectedRate != null &&
+                              _selectedRate >= counter / 2)
+                          ? Colors.white
+                          : Colors.white30,
+                    )),
+              ],
+            ),
+          ),
+        ],
+      ),
+    );
+  }
+
+  BorderRadius _roundedBorder(int index, int total) {
+    if (index == 0) {
+      return BorderRadius.only(
+          topLeft: Radius.circular(10), bottomLeft: Radius.circular(10));
+    } else if (index == total - 1) {
+      return BorderRadius.only(
+          topRight: Radius.circular(10), bottomRight: Radius.circular(10));
+    } else {
+      return null;
+    }
```

Comment:
It's easier to read IMO 🤔 
```suggestion
    if (index == 0) {
      return BorderRadius.only(
          topLeft: Radius.circular(10), bottomLeft: Radius.circular(10));
    }
    
    if (index == total - 1) {
      return BorderRadius.only(
          topRight: Radius.circular(10), bottomRight: Radius.circular(10));
    }
    
    return null;
```